src/Nethermind/Nethermind.Merge.Plugin/Synchronization/ChainLevelHelper.cs

```diff
@@ -100,45 +103,28 @@ public class ChainLevelHelper : IChainLevelHelper
         return headers.ToArray();
     }
 
-    public Block[]? GetNextBlocks(int maxCount)
+    public void SetNextBlocks(int maxCount, BlockDownloadContext context)
```

Comment:
NP: Recommend the term "Populate" instead of "Set" here. Or put this method in `BlockDownloadContext` instead i guess.